### DISCUSSION TITLE
Allow Sections to be Hidden from Entries Section Sidebar

### DISCUSSION THREAD

**Discussion by kylecotter** [craftcms/cms#11321] [Ideas] [answered]
Usage: when creating sections that are purely for "component" entries - e.g for use primary with the new slide-out UI as a related entry within an entry, it'd be great to be able to hide those from the main sections list.

e.g. 
![image](https://user-images.githubusercontent.com/505894/170360958-27b553b8-e56d-47eb-a469-294d66a7fa11.png)

**Comment by michaelrog** [comment]
I imagine you could accomplish this pretty trivially via [`EVENT_REGISTER_SOURCES`](https://docs.craftcms.com/api/v3/craft-events-registerelementsourcesevent.html). (Feel free to ping me if you'd like help with it!)

**Comment by brandonkelly** [comment, marked as answer]
Already possible as of 4.0 :)

From the “Customize…” modal, you can disable each of the native sources now. And headings will only be visible if they are followed by at least one visible source.

**Reply by kylecotter** [reply]
@brandonkelly following up on this one, is it intentional that when a native source is disabled from that Customize modal for Element Indexes that you also _can't_ create entries in that section from a Related Entries Fieldtype modal?

I can see that being useful in some contexts, but also not in others (e.g. the "Component" section philosophy this isn't helpful) .

![image](https://user-images.githubusercontent.com/505894/171966465-89b4fc97-373a-4569-92b0-73e1b7348aa4.png)

**Reply by brandonkelly** [reply]
@kylecotter That’s probably happening because there’s no available sources for the modal, which prevents the embedded element index from fully initializing, since it relies on having at least once source.

https://github.com/craftcms/cms/blob/623f8899b25aa957be7ec1f59cad7a11e0599499/src/web/assets/cp/src/js/BaseElementIndex.js#L417-L420

**Comment by kylecotter** [comment]
Geez, thanks @brandonkelly. Was staring right at it. Not sure why `Enabled` didn't register for me. Thanks!
